### PR TITLE
980 feature delete environment variables

### LINKED ISSUE #980: [FEATURE] Delete environment variables
Add a delete button to the environment variable. Similar to delete capability and talent.

### PR DESCRIPTION
**Pull request by minump**
## Description
Added a close button to remove env key-value pair in catalog create and edit page.

Make environmentVariable key required if present.
In the contribute.html data_validation() method (which is called in submit button), get all elements with the class name envvar-key. Iterate through the list and check if any is empty. Return False if empty.
In contribution/api/dataset_utils, check if key is non-empty.

Merged [PR#1031](https://github.com/rokwire/rokwire-building-blocks-api/pull/1031) with this PR.

**Fixes #980, #1028**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: contributions/catalog/webapps/templates/contribute/contribute.html [hunk 1]
```diff
@@ -592,7 +592,8 @@ <h3 id="capability">Add a Capability to this Contribution</h3>
                                 <input class="form-control envvar-value" name="environmentVariables_value_0" placeholder="Value"
                                        id="environmentVariables_value_0" type="password">
                                 <i class="bi bi-eye-slash field_icon" id="togglePassword"></i>
-                                <br>
+                                <i class="bi bi-x-lg" id="envvarclose"></i>
```
Comment (sandeep-ps) [inline contributions/catalog/webapps/templates/contribute/contribute.html:595]: Let's use hyphenated words for element ID names and CSS class names. So, something like `env-var-close` or `env-var-close-btn` may be better. Thanks.

Comment (minump) [reply contributions/catalog/webapps/templates/contribute/contribute.html:595]: updated

**Comment by minump** [comment]
merged #1028 with this PR.

File: contributions/catalog/webapps/templates/contribute/contribute.html (hunk 1 above)
Comment (sandeep-ps) [reply contributions/catalog/webapps/templates/contribute/contribute.html]: Thanks.

**Review by bingzhang — APPROVED** [review]

**Review by sandeep-ps — APPROVED** [review]
Thanks. The CSS class/ID name changes look good. Approving.

**Comment by minump** [comment]
merging with develop

**Pull request merged**